test/e2e/framework/pv/pv.go

```diff
@@ -844,3 +848,21 @@ func WaitForPersistentVolumeDeleted(c clientset.Interface, pvName string, poll,
 	}
 	return fmt.Errorf("PersistentVolume %s still exists within %v", pvName, timeout)
 }
+
+// WaitForPVCFinalizer waits for a finalizer to be added to a PVC in a given namespace.
+func WaitForPVCFinalizer(ctx context.Context, cs clientset.Interface, name, namespace, finalizer string, poll, timeout time.Duration) error {
+	defer ginkgo.GinkgoRecover()
+	if successful := utils.WaitUntil(poll, timeout, func() bool {
+		pvc, err := cs.CoreV1().PersistentVolumeClaims(namespace).Get(ctx, name, metav1.GetOptions{})
+		framework.ExpectNoError(err, fmt.Sprintf("failed to get pvc %s/%s", namespace, name))
```

Comment:
This should probably not fail if it's a not found error?

Reply:
Should we let the caller of `WaitForPVCFinalizer` to decide what to do with a `not found` error?  In this particular e2e test case, we are trying to delete PVC being used as a source to take snapshot before snapshot is ready.  So we need to make sure that this PVC finalizer is added to the PVC before we try to delete PVC.  If PVC is not found, that means something outside of this test is deleting the PVC ahead of time.  That will cause this particular test to fail any way because snapshot ReadyToUse field will be forever `false` so that create snapshot can never succeed.

Reply:
Maybe we should print a message if PVC is not found.

Reply:
Modified this to let the caller decide what to do with _any_  error. If `Get()` fails, the function will retry until `timeout` occurs and the callers will receive the last error.

Reply:
> Modified this to let the caller decide what to do with any error

Should we leave the error unwrapped if we wanted to enable this?